app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,84 @@
+require 'forwardable'
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Err = Errors::Authentication::AuthnAzure
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                ::Role,
+            resource_class:             Resource,
+            application_identity_class: ApplicationIdentity
+        },
+        inputs:       %i(account, service_id, username, xms_mirid, oid)
+    ) do
+
+      def call
+        parse_token_field
+        populate_identity_values
+        # compare xms_mirid with what is defined in annotations
+        validate_role_annotations
+        # check integrity of annotations
+        validate_application_annotations
+      end
+
+      private
+
+      # traverse xms_mirid field for later value comparisons
+      def parse_token_field
+        field_split = xms_mirid.split('/')
+        @xms_mirid_hash = Hash[field_split.each_slice(2).to_a]
+      end
+
+      # populates xms_mirid object representation with received xms_mirid values
+      def populate_identity_values
+        @actual_xms_mirid = {
+            subscription_id:  @xms_mirid_hash["subscriptions"],
+            resource_group:   @xms_mirid_hash["resourcegroups"]
+        }
+
+        if @xms_mirid_hash["providers"] == "Microsoft.ManagedIdentity"
+          @actual_xms_mirid["userAssignedIdentities"] = @oid
+        elsif xms_mirid_hash["providers"] == "Microsoft.Compute"
+          @actual_xms_mirid["system_assigned_identity"] = @xms_miridHash["virtualMachines"]
```

Comment:
do you think it would be appropriate to just supply an `else` in this conditional block because I don't believe we support anything that is not `Microsoft.ManagedIdentity` or `Microsoft.Compute`

Reply:
That's correct. Such an `else` clause will raise an error. You can use the diagram from the design to verify you have the correct `if` tree